packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/event_client/string_database/dart_string_database.stub.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:async';
+
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/string_database/string_database.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/string_database/string_database_element.dart';
+import 'package:aws_common/aws_common.dart';
+import 'package:drift/drift.dart';
+
+/// {@template amplify_analytics_pinpoint_dart.dart_string_database}
+/// Stores strings using IndexDB for web and Drift for all other platforms
+/// {@endtemplate}
+class DartStringDatabase implements StringDatabase, Closeable {
```

Comment:
I don't get what a "string" database is... how about `EventDatabase` for the interface and `PlatformEventDatabase` for this wrapper class?

I feel like these methods should accept `Event` objects as well and be named appropriately, e.g. `saveEvent`, `deleteEvents`, `listEvents`

Reply:
Hey this is how things are working now. 

The `EventDatabase` you suggest is the current `EventStorageAdapter` which also accepts `Event` objects and already has `saveEvent` `retrieveEvents` and `deleteEvents` methods. 

This is how storage works:

`EventStorageAdapter.saveEvent(Event event)` -> 
`DartStringDatabase.addString`( event encoded as json string) 

The `DartStringDatabase` is an internal class that just handles the storage of strings.  `EventStorageAdapter` deals with converting stored `Event` as strings and retrieving those strings and turning them back to `Event.

Reply:
I think Dillon might be noting that the name doesn't help give context as to what this is/does. A "String Database" is not term people would be familiar with, nor does it really describe what this does or what it is intended to be used for.

EventDatabase would indicate it is used to store events. It sounds like you intentionally wanted to keep a level of abstraction between this DB and Analytics events though, which I think is fine.

This does indeed store Strings, but so do other storage mechanisms we have, such as secure_storage. What makes this different? This data store has some unique characteristics. It stores objects (that happen to be strings) in a First in, First out (FIFO) manner. The delete method is an exception here, as it accepts specific IDs to delete, rather than deleting the first N items.

Naming is hard. We aren't going to find a perfect name to describe this unique data store, but I think we can do better than StringDatabase/StringDatabaseElement.

Below is one option. In addition to the name change, I updated the method names, and changed the API signature of delete to take items, instead of just Ids. If `StoredEvent` were to extend/implement FIFODataStoreItem, you could just pass in the StoredEvents to deleteEvents, and let this class be concerned with how to delete those items. It would help maintain the level of abstraction / separation of concerns that I think you wanted achieve. 

```dart
/// A Data store that inserts and retrieves data in a FIFO (First In, First Out)
/// manner.
abstract class FIFODataStore {
  /// Insert an item into the store.
  FutureOr<void> insert(String item);

  /// Get the first [count] items.
  FutureOr<Iterable<FIFODataStoreItem>> get(int count);

  /// Delete the [items] from the store.
  FutureOr<void> delete(Iterable<FIFODataStoreItem> items);

  /// Delete all items in database.
  FutureOr<void> clear();

  /// Closes this database and releases associated resources.
  FutureOr<void> close();
}

/// An item stored in the [FIFODataStore].
class FIFODataStoreItem {
  const FIFODataStoreItem({required this.id, required this.value});

  /// The ID in the database.
  final int id;

  /// The value of the item.
  final String value;
}
```

Reply:
Agree with above, I think naming should indicate what's the database capable of, but not what format it stores "things" in. TBH I was very confused by StringDatabase when I just started looking at this PR 😅

Reply:
Thanks guys that sounds good, I've updated the PR to have the new naming scheme / API that Jordan suggested.